plugins/cue_points/quiz/services/QuizUserEntryService.php

```diff
@@ -19,25 +19,67 @@ class QuizUserEntryService extends KalturaBaseService{
 	 */
 	public function submitQuizAction($id)
 	{
+		KalturaLog::debug("Submit Quiz for QuizUserEntryId [" .$id."]");
 		$dbUserEntry = UserEntryPeer::retrieveByPK($id);
 		if (!$dbUserEntry)
-			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $id);
+		{
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $id);
+		}
 
 		if ($dbUserEntry->getType() != QuizPlugin::getCoreValue('UserEntryType',QuizUserEntryType::QUIZ))
 		{
-			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_TYPE, $dbUserEntry->getType());
+			throw new KalturaAPIException(KalturaQuizErrors::PROVIDED_ENTRY_IS_NOT_A_QUIZ, $id);
 		}
+
 		/**
-		 * @var QuizUserEntry $dbUserEntry
-		 */
+		* @var QuizUserEntry $dbUserEntry
+		*/
 		$score = $dbUserEntry->calculateScore();
 		$dbUserEntry->setScore($score);
-//		$dbUserEntry->setStatus(QuizUserEntryStatus::QUIZ_SUBMITTED);
+		KalturaLog::debug("Quiz score is [" .$score."]");
 		$dbUserEntry->setStatus(QuizPlugin::getCoreValue('UserEntryStatus', QuizUserEntryStatus::QUIZ_SUBMITTED));
 		$dbUserEntry->save();
 
 		$userEntry = new KalturaQuizUserEntry();
 		$userEntry->fromObject($dbUserEntry, $this->getResponseProfile());
+
+		$entryId = $dbUserEntry->getEntryId();
+		$entry = entryPeer::retrieveByPK($entryId);
+		if(!$entry)
+		{
+			throw new KalturaAPIException(KalturaErrors::INVALID_OBJECT_ID, $entryId);
```

Comment:
I wouldn't throw exception after the actual save already done, the use may think that the quiz was not submitted.